applications/tari_base_node/src/builder.rs

```diff
@@ -473,19 +471,59 @@ where
         .set_base_node_public_key(base_node_public_key.clone())
         .await
         .expect("Problem setting local base node public key for transaction service.");
+    // Start the Output Manager UTXO Validation
+    let state_machine = base_node_handles
+        .get_handle::<StateMachineHandle>()
+        .expect("Could not get State Machine handle");
     let mut oms_handle = wallet_handles
         .get_handle::<OutputManagerHandle>()
         .expect("OutputManagerService is not registered");
-    oms_handle
-        .set_base_node_public_key(base_node_public_key)
-        .await
-        .expect("Problem setting local base node public key for output manager service.");
-    // Start the Output Manager UTXO Validation
-    oms_handle
-        .validate_utxos(UtxoValidationRetry::UntilSuccess)
-        .await
-        .expect("Problem starting the Output Manager Service Utxo Valdation process");
-
+    task::spawn(async move {
+        let mut status_watch = state_machine.get_status_info_watch();
+        loop {
+            let bootstrapped = match status_watch.recv().await {
+                None => false,
+                Some(s) => match s {
+                    StatusInfo::Listening(li) => li.is_bootstrapped(),
+                    _ => false,
+                },
+            };
+            if bootstrapped {
+                debug!(
+                    target: LOG_TARGET,
+                    "Initial sync achieved and starting with UTXO validation process.",
+                );
+                oms_handle
+                    .set_base_node_public_key(base_node_public_key)
+                    .await
+                    .map_err(|e| {
+                        error!(
+                            target: LOG_TARGET,
+                            "An error occurred with Output Manager Service setting the base node public key: {}", e
+                        );
+                        e
+                    })
+                    .expect("An error occurred with Output Manager Service setting the base node public key.");
+                oms_handle
+                    .validate_utxos(UtxoValidationRetry::UntilSuccess)
+                    .await
+                    .map_err(|e| {
+                        error!(
+                            target: LOG_TARGET,
+                            "An error occurred with Output Manager Service starting the UTXO Validation process: {}", e
+                        );
+                        e
+                    })
+                    .expect("An error occurred with Output Manager Service starting the UTXO Validation process");
+                break;
+            }
+            debug!(
+                target: LOG_TARGET,
+                "UTXO validation process not started while busy with initial sync.",
+            );
+            delay_for(Duration::from_secs(30)).await;
```

Comment:
This is actually unnecessary, the recv() above will block until a new status is published.

Reply:
Updated